docs/setup.rst

```diff
@@ -94,18 +95,97 @@ The following sub-sections provide guidance on how to install the recommended to
+
+  mkdir build-zlib
+  cd build-zlib
+
+Create the zlib library build instructions using CMake::
+
+  cmake -DCMAKE_INSTALL_PREFIX=<ZLIB_INSTALL_DIR> ..\zlib
+
+Where <ZLIB_INSTALL_DIR> is to be replaced by a path on your computer.  Something like::
+
+  C:\Users\Andre\libraries
+
+if your username was 'Andre'.
+
+We can now build the library with the command::
+
+  cmake --build . --config Release
+
+The last thing we do is install the library with the command::
+
+  cmake --build . --target install --config Release
+
 .. _setup_libxml2:
 
 LibXml2
 -------
-:libxml2:`LibXml2<>` is a parser and toolkit for manipulating XML files and text.  
-It's important to use a 64-bit version of the library as the 32-bit is not compatible with libCellML.  
+:libxml2:`LibXml2<>` is a parser and toolkit for manipulating XML files and text.
+It's important to use a 64-bit version of the library as the 32-bit is not compatible with libCellML.
 
 - **Windows** On Windows, it's easiest to install using the packaged version available from :opencmiss_repo:`the OpenCMISS site<>` .
-  You're welcome to build your own version if you'd rather, but please make sure it's a 64-bit implementation.  
-  Source code for building it yourself can be downloaded from the :libxml2:`LibXml2<>` site.
+  You're welcome to build your own version if you'd rather, but please make sure it's a 64-bit implementation.
+  To build :libxml2:`LibXml2<>` from source refer to the next sub-section.
 - **Linux** LibXml2 can be installed using :code:`sudo apt install libxml2-dev`.
-- **macOS** LibXML2 is already installed on macOS, so no further action is required on that platform.  
+- **macOS** LibXML2 is already installed on macOS, so no further action is required on that platform.
+
+Building from source
+++++++++++++++++++++
+
+To create a LibXml2 library suitable for libCellML we need to download, configure, build, and install it.
+LibXml2 can be configured to use zlib so build and install that first before following these instructions.
+
+To download LibXml2 clone the repository https://github.com/OpenCMISS-Dependencies/libxml2::
+
+  git clone https://github.com/OpenCMISS-Dependencies/libxml2 -b v2.9.10
+
+Create a build directory and change into it::
+
+  mkdir build-libxml2
+  cd build-libxml2
+
+Create the LibXml2 library build instructions using CMake::
+
+  cmake -DCMAKE_PREFIX_PATH=<ZLIB_INSTALL_DIR> -DCMAKE_INSTALL_PREFIX=<LIBXML2_INSTALL_DIR> -DLIBXML2_WITH_DEBUG=OFF -DLIBXML2_WITH_DOCB=OFF -DLIBXML2_WITH_FTP=OFF -DLIBXML2_WITH_ICONV=OFF -DLIBXML2_WITH_ICU=OFF -DLIBXML2_WITH_LZMA=OFF -DLIBXML2_WITH_MEM_DEBUG=OFF -DLIBXML2_WITH_PROGRAMS=OFF -DLIBXML2_WITH_PYTHON=OFF -DLIBXML2_WITH_RUN_DEBUG=OFF -DLIBXML2_WITH_SAX1=OFF -DLIBXML2_WITH_TESTS=OFF ..\libxml2
+
+Where <ZLIB_INSTALL_DIR> is to be replaced by the path on your computer where zlib is installed.
+<LIBXML2_INSTALL_DIR> is to be replaced by a path on your computer.
+Using the same directory that the zlib library is a good idea.
```

Comment:
Why mention this? It sounds like it's "ok" if people don't use the same directory, so why specify this?

Reply:
It is okay for people to use a different directory and there is no tangible difference if you do.  As far as these instructions are concerned it is a good idea to use the same directory.

Reply:
I don't see the point of "instructions" that are not needed per se. It's already long enough as it is, so no point in my view to make it unnecessarily longer. Anyway...